lib/internal/Magento/Framework/App/Test/Unit/Console/CommandLoader/AggregateTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Magento\Framework\App\Test\Unit\Console\CommandLoader;
+
+use Magento\Framework\Console\CommandLoader\Aggregate;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\CommandLoader\CommandLoaderInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+class AggregateTest extends TestCase
+{
+    private $firstCommandLoader;
+    private $secondCommandLoader;
+    private $subj;
+
+    protected function setUp(): void
+    {
+        $this->firstCommandLoader = $this->getMockBuilder(CommandLoaderInterface::class)->getMock();
+        $this->secondCommandLoader = $this->getMockBuilder(CommandLoaderInterface::class)->getMock();
+        $this->subj = new Aggregate([$this->firstCommandLoader, $this->secondCommandLoader]);
+    }
+
+    /**
+     * @dataProvider provideTestCasesForHas
+     */
+    public function testHas($firstResult, $secondResult, $overallResult)
```

Comment:
The test method name should be more descriptive about what it tests, what is the expected outcome.

Reply:
I spent some time thinking about how to make this name better, and I'm not sure what else to add. Because it uses a data provider, it's meant to cover every known case of the method.

I could switch strategies and build multiple test methods: `testHasWithAtLeastOneMatch`, `testHasWithZeroMatches`, etc. but I'm not sure that's better (aside from having descriptive names). I did add significant prose to the docblock, which I believe helps explain what the test achieves.